src/exla/exla_nif_util.cc

```diff
@@ -52,7 +52,31 @@ namespace exla {
 
   int get_options(ErlNifEnv* env, ERL_NIF_TERM term, xla::ExecutableRunOptions &options){ return 0; }
   int get_options(ErlNifEnv* env, ERL_NIF_TERM term, xla::ExecutableBuildOptions &options){ return 0; }
-  int get_options(ErlNifEnv* env, ERL_NIF_TERM term, xla::LocalClientOptions &options){ return 0; }
+  int get_options(ErlNifEnv* env, ERL_NIF_TERM term, xla::LocalClientOptions &options){
+    size_t map_size;
+
+    if(!enif_get_map_size(env, term, &map_size) || map_size != 4) return 0;
+
+    ERL_NIF_TERM platform_arg, number_of_replicas_arg, intra_op_parallelism_threads_arg
+
+    if(!enif_get_map_value(env, term, exla::make(env, "platform"), &platform_arg)) return 0;
+    if(!enif_get_map_value(env, term, exla::make(env, "number_of_replicas"), &number_of_replicas_arg)) return 0;
+    if(!enif_get_map_value(env, term, exla::make(env, "intra_op_parallelism_threads"), &intra_op_parallelism_threads_arg)) return 0;
+
+    stream_executor::Platform* platform;
+    int number_of_replicas, intra_op_parallelism_threads;
+
+    // TODO: Adjust this to accept `int` which is passed from Elixir
+    if(!exla::get_platform(env, platform_arg, platform)) return 0;
+    if(!exla::get(env, number_of_replicas_arg, number_of_replicas)) return 0;
+    if(!exla::get(env, intra_op_parallelism_threads_arg, intra_op_parallelism_threads)) return 0;
```

Comment:
Similar to the other discussions, should we simply call the NIF as: `EXLA.NIF.get_client(number_of_replicas, intra_op_parallelism_threads, ...)` instead of passing a map? This would allow us to normalize and validate the data within Elixir instead.

Reply:
So rather in `Exla.Client.create_client` do:

```elixir
def create_client(options = %LocalClientOptions{}) do
   Exla.NIF.get_or_create_local_client(options.device, options.replicas, ....)
```

I see what you mean, this will simplify the C++ implementations as well